src/connector.ts

```diff
@@ -200,14 +205,24 @@ export class Connector {
           privateKey &&
           serverCaCert
         ) {
-          return getSocket({
+          const tlsSocket = getSocket({
             instanceInfo,
             ephemeralCert,
             host,
             port,
             privateKey,
             serverCaCert,
           });
+          tlsSocket.once('secureConnect', async () => {
+            // TLS handshake failed
+            if (!tlsSocket.authorized) {
+              await cloudSqlInstance.forceRefresh();
+            }
+          });
+          tlsSocket.once('error', async () => {
+            await cloudSqlInstance.forceRefresh();
+          });
```

Comment:
is there a specific reason we need to differentiate between these two different actions instead of just having one catch all?

Reply:
That is the way to handle both these events in Node.js as the TLS Handshake _may_ not emit an error event, and should be handled in `secureConnect` instead.

Reply:
I would like to point out 2 things:
* I believe that comment `// TLS handshake failed` is misleading. The TLS handshake at this point did in fact succeed (`secureConnect` event is emitted on successful TLS handshake). However `authorized === false` means that server certificate could not be verified by CA.
* [option `rejectUnathorized = true`](https://nodejs.org/docs/latest-v14.x/api/tls.html#tls_tls_connect_options_callback) by default, and should emit `error` event if server certificate could not be verified by CA. Therefore I believe that the first handler could be unnecessary.

Reply:
you're right, I'll remove the `secureConnect` handler for now. Ideally we should have an e2e scenario that we can tweak to exercise a bit more the diff ways this may fail in order to trigger a force refresh.